packages/ui/src/providers/SearchParams/index.tsx

```diff
@@ -17,20 +19,20 @@ const initialContext: SearchParamsContext = {
 
 const Context = createContext(initialContext)
 
-function createParams(search: string) {
+export function createParams(params: ReadonlyURLSearchParams): State {
+  const search = params.toString()
+
   return qs.parse(search, {
     depth: 10,
     ignoreQueryPrefix: true,
   })
 }
 
-// TODO: abstract the `next/navigation` dependency out from this provider so that it can be used in other contexts
+// TODO: this provider should likely be marked as deprecated and then removed in the next major release
```

Comment:
Should we use the `@deprecated` JSDoc and as well on `useSearchParams`?

Reply:
Though it's still used, maybe not!

Reply:
I thought about it, but I wasn't quite ready to commit to that change yet. Might be best though. I will add now.

Reply:
Same with the params provider, except that's not used at all internally. Should not even exist.

Reply:
Going to address both of these together in another PR because it will require some code migration to do right.